test/e2e/policies.test-suite/balloons/n4c16/test22-isolcpus/code.var.sh

```diff
@@ -0,0 +1,59 @@
+ns=isolcpus
+
+reboot-node() {
+    vm-reboot
+    timeout=3600 host-wait-vm-ssh-server $OUTPUT_DIR
+}
+
+restart-kubelet() {
+    vm-command "systemctl restart kubelet"
+    sleep 5
+    vm-wait-process --timeout 120 kube-apiserver
+    vm-command "cilium status --wait --wait-duration=120s --interactive=false"
+}
+
+vm-command "grep isolcpus=0,1 /proc/cmdline" || {
+    if [[ "$distro" == *fedora* ]]; then
+        fedora-set-kernel-cmdline "isolcpus=0,1"
+    else
+        ubuntu-set-kernel-cmdline "isolcpus=0,1"
+    fi
+    reboot-node
+    vm-command "grep isolcpus=0,1 /proc/cmdline" || {
+        error "failed to set isolcpus kernel commandline parameter"
+    }
+    restart-kubelet
+}
+
+helm-terminate
+helm_config=${TEST_DIR}/balloons-isolcpus.cfg helm-launch balloons
+vm-command "kubectl create namespace $ns"
+
+# pod0: runs on system isolated CPUs
+CONTCOUNT=2 namespace="$ns" create balloons-busybox
+report allowed
+verify "cpus['pod0c0'] == {'cpu00', 'cpu01'}"
+
+# pod1: should run on non-isolated CPUs
```

Comment:
There's one essential thing that this test should test:
**when** new balloons are created (or existing balloons inflated) **and** there are isolcpus available in the system,
**is it possible that** an isolcpu could end up into a balloon that does not prefer them.

Currently both isolcpus seem to be already allocated into the "isolcpus" balloon (based on line 35), and therefore it cannot happen that isolcpus would be allocated into new balloons (or inflated existing balloons).

Would you create pods in a different order? First create containers that need most (but not all non-isolcpus) and verify that none of isolcpus was used for that. That is, first create containers that need many CPUs and verify something like: `verify "set.union(cpus['pod0c0'], cpus['pod0c1'], ...).isdisjoint({'cpu00', 'cpu01'})"`

Reply:
Fixed.